src/magic_folder/common.py

```diff
@@ -33,6 +35,29 @@ class BadMetadataResponse(Exception):
     cannot be interpreted as that metadata.
     """
 
+@attr.s(auto_exc=True, frozen=True)
+class APIError(Exception):
+    """
+    An error to be reported from the API.
+    """
+
+    reason = attr.ib(validator=attr.validators.instance_of(unicode))
+    code = attr.ib(
+        default=None,
+        validator=attr.validators.optional(attr.validators.instance_of(int)),
+    )
+
+    @classmethod
+    def from_exception(cls, code, exception, prefix=None):
```

Comment:
should document the params. I assume "code" is an HTTP status code, but would be good to say that (if it's true).